File: university-work/before/liceu/clsx/div&imp/quicksor.pas
Program de_aranjare_a_unui_sir_in_ordine_crescatoare_QuickSort;
uses crt;
type vector=array[1..100] of integer;
var n,a,k:byte;
    x:vector;

Procedure poz(s,d:integer;var k:byte);
var y,aux:integer;
begin
 y:=x[(s+d) div 2];
 repeat
  while x[s]<y do s:=s+1;
  while x[d]>y do d:=d-1;
  if s<=d then
           begin
            aux:=x[s];
            x[s]:=x[d];
            x[d]:=aux;
           end;
 until s=d;
 k:=s;
end;

Procedure quick(s,d:integer);
begin
 if s<d then
  begin
   poz(s,d,k);
   quick(s,k-1);
   quick(k+1,d);
  end;
end;

begin
 clrscr;
 writeln;
 writeln('Acesta este un program care daca ii veti introduce un sir de nume-');
 writeln('re vi le va ordona pe acestea crescator prin metoda "Quick Sort"');
 writeln;
 write('Numarul de elemente din sir : ');readln(n);
 for a:=1 to n do
  begin
   write('Elementul ',a,' : ');
   readln(x[a]);
  end;
 quick(1,n);
 for a:=1 to n do
  write(x[a]:3);
 readkey;
end.

File: university-work/before/c/div&imp/quicksor.cpp
/* sortarea rapida */
#include <stdio.h>
#include <conio.h>
int k;

poz(int li, int ls, int *k, int a[100])
{
   int i, j, i1, j1, aux;

   i= li;
   j= ls;
   i1= 0;
   j1= -1;
   while (i < j)
   {
      if (a[i] > a[j])
      {
	 aux= a[i];
	 a[i]= a[j];
	 a[j]= aux;
	 aux= i1;
	 i1= -j1;
	 j1= -aux;
      }
      i= i+i1;
      j= j+j1;
      if (i == j)
	 *k= i;
   }

   return 0;
}

quick(int li, int ls, int a[100])
{
   if (li < ls)
   {
      poz(li, ls, &k, a);
      quick(li, k-1, a);
      quick(k+1, ls, a);
   }

   return 0;
}

main()
{
   int n, i, a[100];

   clrscr();
   printf("\n");

   printf(" Numarul de elemente din vector : ");
   scanf("%i", &n);
   for (i= 1; i <= n; i= i+1)
   {
      printf("  Elementul %i : ", i);
      scanf("%i", &a[i]);
   }
   quick(1, n, a);
   printf(" Sirul ordonat este : ");
   for (i= 1; i <= n; i= i+1)
      printf("%i ", a[i]);

   getch();
   return 0;
}